sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -1192,18 +1192,19 @@ private[lf] object Speedy {
           val missing = arity - actualsSoFar.size
           val newArgsLimit = Math.min(missing, newArgs.length)
 
-          val actuals = new util.ArrayList[SValue](actualsSoFar.size + newArgsLimit)
-          discard[Boolean](actuals.addAll(actualsSoFar))
+          val actuals = Array.ofDim[SValue](actualsSoFar.size + newArgsLimit)
+          Array.copy(actualsSoFar, 0, actuals, 0, actualsSoFar.size)
 
           val othersLength = newArgs.length - missing
 
+          // Evaluate the arguments
           // Evaluate the arguments
           var i = 0
+          var j = actualsSoFar.size
           while (i < newArgsLimit) {
```

Comment:
Without looking at any of the other code (beyond this diff), you sort of feel this while can be turned into  for-loop. As now one has ever done this, I presume there's an obvious reason why that I'm not currently seeing?

Reply:
Oh, that's interesting. Thanks for that.